### PR TITLE
doc: add upgrade guide from 5.2 to 2023.1

### PR DESCRIPTION
**Pull request by annastuchlik**
Related: https://github.com/scylladb/scylla-enterprise/issues/2770

This commit adds the upgrade guide from ScyllaDB Open Source 5.2 to ScyllaDB Enterprise 2023.1.
This commit does not cover metric updates (the metrics file has no content, which needs to be added in another PR).

As this is an upgrade guide, this commit must be merged to master and backported to **branch-5.2** and **branch-2023.1** in scylla-enterprise.git.

### CODE REVIEW

**Comment by kbr-scylla** [comment]
> As this is an upgrade guide, this commit must be merged to master and backported to branch-5.2 and branch-2023.1 in scylla-enterprise.git.

Is it desirable that the upgrade guide will also show up in the open source documentation - since it involves an enterprise release?
If not, perhaps we should open this PR directly against scylla-enterprise repo?

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+.. _upgrade-debian-ubuntu-5.2-to-enterprise-2023.1: 
+
+Download and install the new release
+------------------------------------
+
+.. tabs::
+
+   .. group-tab:: Debian/Ubuntu
+
+        Before upgrading, check what version you are running now using ``dpkg -s scylla-server``. You should use the same version as this version in case you want to |ROLLBACK|_ the upgrade. If you are not running a |SRC_VERSION|.x version, stop right here! This guide only covers |SRC_VERSION|.x to |NEW_VERSION|.y upgrades.
```
Comment (fgelcer) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst:122]: why not `scylla --version`?

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+               sudo apt-get update
+               sudo apt-get remove scylla\*
+               sudo apt-get install scylla-enterprise
+               sudo systemctl daemon-reload
+
+        Answer ‘y’ to the first two questions.
+
+   .. group-tab:: RHEL/CentOS
+
+        Before upgrading, check what version you are running now using ``rpm -qa | grep scylla-server``. You should use the same version as this version in case you want to |ROLLBACK|_ the upgrade. If you are not running a |SRC_VERSION|.x version, stop right here! This guide only covers |SRC_VERSION|.x to |NEW_VERSION|.y upgrades.
```
Comment (fgelcer) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst:154]: why not `scylla --version`?

File: docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+            .. code:: sh
+
+               sudo yum clean all
+               sudo rm -rf /var/cache/yum
+               sudo yum remove scylla\*
+               sudo yum install scylla-enterprise
+
+   .. group-tab:: EC2/GCP/Azure Ubuntu Image
+
+        Before upgrading, check what version you are running now using ``dpkg -s scylla-server``. You should use the same version as this version in case you want to |ROLLBACK|_ the upgrade. If you are not running a |SRC_VERSION|.x version, stop right here! This guide only covers |SRC_VERSION|.x to |NEW_VERSION|.y upgrades.
```
Comment (fgelcer) [inline docs/upgrade/upgrade-to-enterprise/upgrade-guide-from-5.2-to-2023.1/upgrade-guide-from-5.2-to-2023.1-generic.rst:170]: why not `scylla --version`?

**Comment by annastuchlik** [comment]
I've applied the feedback from @fgelcer. This PR is now ready to be merged.

**Comment by denesb** [comment]
> What's the process for that? If a commit is backported to branch-5.2, is it automatically backported to branch-2023.1 (as 2023.1 is built on 5.2)? Or is it a manual process?

Yes :). In general, any patch that gets into branch-X in OSS, will also get "back-merged" into the enterprise release which is based on said branch. The caveat is that this happens only until said OSS release is EOL'd. After that, backports targetting an enterprise release which has an EOL'd OSS base branch, must be backported to the enterprise branch directly.
This is a semi-automatic process, it is done by maintainers (Avi mostly).

> What should I do to make sure that an update I make is present on a given branch in the scylla-enterprise repository?

Just ask for a backport to the OSS branch that is its base . Unless the change is specific to enterprise, in which case ask for it to be backported to enterprise only.

**Comment by annastuchlik** [comment]
That's awesome :) @denesb Thanks for the explanation! No more questions (for now :smiling_imp:).

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to branch-5.2.

**Pull request closed without merging**